Reject certificates with explicit ec parameters in strict mode [Backport Upstream]

hustliyilin opened the pull request:
These patches both from `openssl` upstream and redhat `openssl-1.1.1.src.rpm`.  I backport and rebase babassl 8.2.1 to reject certificates with explicit ec parameters in strict mode.

InfoHunter commented:
> I backport and rebase babassl 8.2.1 to reject certificates with explicit ec parameters in strict mode.

If you intended to backport this to 8.2.1, you need to target the branch to 8.2-stable instead of master. (Same for your previous patches as well)

hustliyilin commented:
>  intended to backport this to

Thanks for you suggestion. I intended to backport this to the master branch, I will rebase master and re-push it.

hustliyilin commented:
According to your suggestion, I modified the title of this pr, rebased the babassl master branch, and force-pushed these pacthes. Now all CI/CD testcases pass. FYI @InfoHunter
